spartan_parallel/src/lib.rs

```diff
@@ -74,29 +65,29 @@ const W3_WIDTH: usize = 8;
 
 /// `ComputationCommitment` holds a public preprocessed NP statement (e.g., R1CS)
 #[derive(Clone, Serialize)]
-pub struct ComputationCommitment {
-  comm: R1CSCommitment,
+pub struct ComputationCommitment<S: SpartanExtensionField> {
+  comm: R1CSCommitment<S>,
 }
 
 /// `ComputationDecommitment` holds information to decommit `ComputationCommitment`
-pub struct ComputationDecommitment {
-  decomm: R1CSDecommitment,
+pub struct ComputationDecommitment<S: SpartanExtensionField> {
+  decomm: R1CSDecommitment<S>,
 }
 
 /// `Assignment` holds an assignment of values to either the inputs or variables in an `Instance`
 #[derive(Clone, Serialize, Deserialize)]
-pub struct Assignment {
+pub struct Assignment<S: SpartanExtensionField> {
   /// Entries of an assignment
-  pub assignment: Vec<Scalar>,
+  pub assignment: Vec<S>,
 }
 
-impl Assignment {
+impl<S: SpartanExtensionField> Assignment<S> {
```

Comment:
Again: if you rename `S` to be `Scalar` (in most places) it will make your diff so much smaller and easier to review.